lib/services/cacheUtils.js

```diff
@@ -91,10 +97,18 @@ function createDomainEnabledCacheHandler(domain, processValueFn, cache, cacheTyp
 function cacheAndHold(cache, cacheType, cacheKey, retrieveRequestFn, processValueFn, callback) {
     var cachedValue = cache.data[cacheType].get(cacheKey);
 
+    function getCacheEventId() {
+        return cacheType + ':' + cacheKey;
+    }
+
     if (cachedValue[cacheKey]) {
+        logger.debug('Value found in the cache [%s] for key [%s]: %s', cacheType, cacheKey, cachedValue[cacheKey]);
+
         processValueFn(cachedValue[cacheKey], callback);
-    } else if (cache.updating[cacheType]) {
-        cache.channel.on(cacheType,
+    } else if (cache.updating[cacheType][cacheKey]) {
```

Comment:
I see there are no controls on the existence of `cache.updating[cacheType]`. Just wanted to cross-check with you that it will always exist to avoid uncontrolled `ReferenceError`s.